### ISSUE TITLE
multiple labels are shown for one single frame

### ISSUE THREAD

**Issue by monajalal** [jbohnslav/deepethogram#26]
Hi Jim,

We are not sure if this is supposed to be a correct thing but do you also get multiple labels for a single frame? We were hoping that since it is using video information, it only provide one label for each single frame. Here's a result for you to see:

<img width="795" alt="Screen Shot 2020-11-30 at 5 02 21 PM" src="https://user-images.githubusercontent.com/1892917/100676480-fc5aed00-3336-11eb-914c-2e6e0859c88c.png">

Additionally, we are noticing that in our prediction results, there is not a full overlap of labels between the prediction and groundtruth. 

<img width="369" alt="Screen Shot 2020-11-30 at 5 08 36 PM" src="https://user-images.githubusercontent.com/1892917/100676543-1ac0e880-3337-11eb-886b-be0bd2b4252e.png">

Here the first line is the groundtruth and the second line is prediction. Like you see class 1 and 0 do not overlap. 

Another problem is predictions are very off.

Could you please provide an answer to the following questions?
1. are we expected to have more than 1 labels for each frame in the prediction results?
2. how can we improve the prediction results?
3. how many minutes of training video do you suspect would be enough if we have like 10 labels for behaviors and we want to get some sensible results? (like at a minimum)

@kubraE

**Comment by jbohnslav** [comment]
This looks pretty interesting! What kind of animal is that?

> there is not a full overlap of labels between the prediction and groundtruth.

I don't know what this means. Can you elaborate? Where did you get those dictionaries?


> 1. are we expected to have more than 1 labels for each frame in the prediction results?

In the paper, we note that all the datasets we used are multi-label, in that there can be multiple behaviors on one single frame. You can one-label-per-frame models with `cfg.feature_extractor.final_activation=softmax`, although there might be bugs here. I couldn't test it well because it did not apply to the datasets I received. I also haven't written this into the sequence models yet. 

> 2. how can we improve the prediction results?

In general, getting more data is the best way. Secondly, if you have very imbalanced datasets, you can try changing the hyperparameter of `train.loss_weight_exp` to something like `0.5`. This corresponds to `$\beta` in the loss calculation in the methods section. Dropout is already pretty high, but you can try to increase data augmentation as well. 

> 3. how many minutes of training video do you suspect would be enough if we have like 10 labels for behaviors and we want to get some sensible results? (like at a minimum)

It's hard to say with any given dataset. The best answer is Figure 5 of the paper: One can get decent performance with only ~100 frames, but of course you'll get better performance with more. After all, 1000 frames is only 16 seconds of behavior at 60fps.

**Comment by jbohnslav** [comment]
Closing due to inactivity, feel free to reopen
